src/main/java/org/jboss/modules/ModuleLoader.java

```diff
@@ -574,13 +575,33 @@ protected final Module findLoadedModuleLocal(String name) {
      * @param module the module to unload
      * @throws SecurityException if an attempt is made to unload a module which does not belong to this module loader
      * @throws SecurityException if the module was not defined by this module loader
+     * @deprecated use {@link #unloadModuleLocal(String,Module)} instead
      */
+    @Deprecated
     protected final void unloadModuleLocal(Module module) throws SecurityException {
+        unloadModuleLocal(null, module);
```

Comment:
For the sake of cleanness of the contract and maintainability, I vote for making the new `unloadModuleLocal()` to accept the module name only rather than name and module. Inside the proper `unloadModuleLocal()` we lookup in the the `moduleMap` anyway.